pkg/io/mdWriter.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package io
+
+import (
+	"bufio"
+	"io"
+	"slices"
+	"strings"
+
+	"github.com/np-guard/vpc-network-config-synthesis/pkg/ir"
+)
+
+const (
+	sgColsNum  = 7
+	aclColsNum = 10
+
+	leftAlign = " :--- "
+	separator = " | "
+)
+
+// MDWriter implements ir.Writer
+type MDWriter struct {
+	w *bufio.Writer
+}
+
+func NewMDWriter(w io.Writer) *MDWriter {
+	return &MDWriter{w: bufio.NewWriter(w)}
+}
+
+func (w *MDWriter) WriteSG(collection *ir.SGCollection, vpc string) error {
+	sgTable, err := WriteSG(collection, vpc)
+	if err != nil {
+		return err
+	}
+	return w.writeAll(slices.Concat(SGHeader(), addAligns(sgColsNum), sgTable))
+}
+
+func (w *MDWriter) WriteACL(collection *ir.ACLCollection, vpc string) error {
+	aclTable, err := WriteACL(collection, vpc)
+	if err != nil {
+		return err
+	}
+	return w.writeAll(slices.Concat(ACLHeader(), addAligns(aclColsNum), aclTable))
+}
+
+func (w *MDWriter) writeAll(rows [][]string) error {
+	for _, row := range rows {
+		if _, err := w.w.WriteString(separator); err != nil {
+			return err
+		}
+		if _, err := w.w.WriteString(strings.Join(row, separator)); err != nil {
+			return err
+		}
+		if _, err := w.w.WriteString(separator + "\n"); err != nil {
+			return err
+		}
```

Comment:
Any reason why not to first build the line, then call `WriteString()` only once?